src/main/scala/junctions/addrmap.scala

```diff
@@ -121,6 +119,12 @@ class AddrMap(
     }.flatten.sortBy(_.region.start)
   }
 
+  flatten.combinations(2) foreach {
+    case (Seq(AddrMapEntry(an, ar), AddrMapEntry(bn, br))) =>
+      require(!(ar.start < br.start + br.size && br.start < ar.start + ar.size),
```

Comment:
After giving this a lot of thought, I finally convinced myself that this expression was correct. Perhaps it would be helpful to add a comment explaining why this expression works.